plugins/catalog/src/components/EntityRelationWarning/EntityRelationWarning.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity, stringifyEntityRef } from '@backstage/catalog-model';
+import {
+  CatalogApi,
+  catalogApiRef,
+  useEntity,
+} from '@backstage/plugin-catalog-react';
+import { Alert } from '@material-ui/lab';
+import React from 'react';
+import useAsync from 'react-use/lib/useAsync';
+import { Box } from '@material-ui/core';
+import { ResponseErrorPanel } from '@backstage/core-components';
+import { useApi, ApiHolder } from '@backstage/core-plugin-api';
+
+async function getRelationWarnings(entity: Entity, catalogApi: CatalogApi) {
+  const entityRefRelations = entity.relations?.map(
+    relation => relation.targetRef,
+  );
+  if (!entityRefRelations || entityRefRelations?.length < 1) {
+    return [];
+  }
+
+  const relatedEntities = await catalogApi
+    .getEntitiesByRefs({ entityRefs: entityRefRelations })
```

Comment:
This should probably have a `fields: ['kind', 'metadata.name', 'metadata.namespace']` to ensure that only the needed parts are actually transfered to the frontend. This can have a distinct impact especially when the number of relations is large.

Might make sense also to check here if the number of relations is very large like more than a thousand), maybe don't perform the call?

Reply:
Implemented the `fields` filter.

I don't know how likely it is to have more than 1000 relations. At least we don't have this, not even over 100 relations.
But I think it makes sense to add this limitation/protection, even when the warning would be especially useful as the error probability with 1000 relations is pretty high 😉